core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -579,4 +580,12 @@ private void checkEmbeddedPicture() {
             hasEmbeddedPicture = Boolean.FALSE;
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof RemoteMedia) {
```

Comment:
this seems wrong. It's not comparing equality on any of the fields in FeedMedia.

Android Studio will easily generate one for you...